extensions/standard-processors/processors/TailFile.cpp

```diff
@@ -403,102 +403,141 @@ void TailFile::onTrigger(const std::shared_ptr<core::ProcessContext> &context, c
     state_file_ = st_file + "." + getUUIDStr();
   }
   if (!this->state_recovered_) {
-    state_recovered_ = true;
-    // recover the state if we have not done so
     this->recoverState(context);
+    state_recovered_ = true;
+  }
+
+  if (tail_mode_ == Mode::MULTIPLE) {
+    checkForRemovedFiles();
+    checkForNewFiles();
   }
 
-  /**
-   * iterate over file states. may modify them
-   */
+  bool did_something = false;
+
+  // iterate over file states. may modify them
   for (auto &state : tail_states_) {
-    auto fileLocation = state.second.path_;
-
-    checkRollOver(context, state.second, state.first);
-    std::string fullPath = fileLocation + utils::file::FileUtils::get_separator() + state.second.current_file_name_;
-    struct stat statbuf;
-
-    logger_->log_debug("Tailing file %s from %llu", fullPath, state.second.currentTailFilePosition_);
-    if (stat(fullPath.c_str(), &statbuf) == 0) {
-      if ((uint64_t) statbuf.st_size <= state.second.currentTailFilePosition_) {
-        logger_->log_trace("Current pos: %llu", state.second.currentTailFilePosition_);
-        logger_->log_trace("%s", "there are no new input for the current tail file");
-        context->yield();
-        return;
+    did_something |= processFile(context, session, state.first, state.second);
```

Comment:
Wonder if we could use std::any_of here

Reply:
`any_of` short-circuits, so I can't use it directly; I could do something like
```c++
  std::vector<bool> did_something(tail_states_.size());
  std::transform(tail_states_.begin(), tail_states_.end(), std::back_inserter(did_something),
      [this, &context, &session](std::map<std::string, TailState>::value_type &state) {
    return processFile(context, session, state.first, state.second);
  });

  if (std::none_of(did_something.begin(), did_something.end(), [](bool x){ return x; })) {
    yield();
  }
```
but I'm not sure that is better.  Would you prefer this?